misk-hibernate/src/main/kotlin/misk/hibernate/HibernateModule.kt

```diff
@@ -44,80 +46,82 @@ class HibernateModule(
-    val sessionFactoryServiceProvider = getProvider(sessionFactoryServiceKey)
+    // Bind PingDataBaseService.
+    bind(keyOf<PingDatabaseService>(qualifier)).toProvider(Provider<PingDatabaseService> {
+      PingDatabaseService(config, environmentProvider.get())
+    }).asSingleton()
+    // TODO(rhall): depending on Vitess is a hack to simulate Vitess has already been started in the
+    // env. This is to remove flakiness in tests that are not waiting until Vitess is ready.
+    // This should be replaced with an ExternalDependency that manages vitess.
+    install(ServiceModule<PingDatabaseService>(qualifier)
+        .dependsOn<StartVitessService>(qualifier))
 
+    // Bind DataSourceService.
     val dataSourceDecoratorsKey = setOfType(DataSourceDecorator::class).toKey(qualifier)
     val dataSourceDecoratorsProvider = getProvider(dataSourceDecoratorsKey)
-
-    val schemaMigratorKey = SchemaMigrator::class.toKey(qualifier)
-    val schemaMigratorProvider = getProvider(schemaMigratorKey)
-
-    val transacterKey = Transacter::class.toKey(qualifier)
-    val transacterProvider = getProvider(transacterKey)
-
-    val pingDatabaseServiceKey = PingDatabaseService::class.toKey(qualifier)
-
-    val dataSourceKey = DataSource::class.toKey(qualifier)
-    val dataSourceProvider = getProvider(dataSourceKey)
-
-    val dataSourceServiceKey = DataSourceService::class.toKey(qualifier)
-
-    bindStartVitessService()
-
-    bind(configKey).toInstance(config)
-
-    bind(pingDatabaseServiceKey).toProvider(Provider<PingDatabaseService> {
-      PingDatabaseService(qualifier, config, environmentProvider.get())
-    }).asSingleton()
-    multibind<Service>().to(pingDatabaseServiceKey)
-
-    // Declare the empty set of decorators
-    newMultibinder<DataSourceDecorator>(qualifier)
-
-    bind(dataSourceKey).toProvider(dataSourceServiceKey).asSingleton()
-    bind(dataSourceServiceKey).toProvider(object : Provider<DataSourceService> {
+    bind(keyOf<DataSource>(qualifier)).toProvider(keyOf<DataSourceService>(qualifier)).asSingleton()
+    bind(keyOf<DataSourceService>(qualifier)).toProvider(object : Provider<DataSourceService> {
       @com.google.inject.Inject(optional = true) var metrics: Metrics? = null
       override fun get() = DataSourceService(
-        qualifier,
-        config,
-        environmentProvider.get(),
-        dataSourceDecoratorsProvider.get(),
-        metrics
+          qualifier,
+          config,
+          environmentProvider.get(),
+          dataSourceDecoratorsProvider.get(),
+          metrics
       )
     }).asSingleton()
-    multibind<Service>().to(dataSourceServiceKey)
+    install(ServiceModule<DataSourceService>(qualifier)
+        .dependsOn<PingDatabaseService>(qualifier))
 
-    bind(sessionFactoryKey).toProvider(sessionFactoryServiceKey).asSingleton()
+    // Bind SessionFactoryService.
+    val entitiesKey = setOfType(HibernateEntity::class).toKey(qualifier)
```

Comment:
this key and `eventListenersKey` can be inlined to stay consistent with the other changes

Reply:
fixed